UnitySoftDebuggerEngine.cs

```diff
@@ -126,27 +130,31 @@ public DebuggerSession CreateSession ()
 			return session;
 		}
 		
-		public ProcessInfo[] GetAttachableProcesses ()
+		public ProcessInfo[] GetAttachableProcesses (Process[] systemProcesses)
 		{
 			int index = 1;
 			List<ProcessInfo> processes = new List<ProcessInfo> ();
-			Process[] systemProcesses = Process.GetProcesses ();
 			StringComparison comparison = StringComparison.OrdinalIgnoreCase;
 			
 			if (null != unityPlayerConnection) {
-				lock (unityPlayerConnection) {
-					foreach (string player in unityPlayerConnection.AvailablePlayers) {
-						try {
-							PlayerConnection.PlayerInfo info = PlayerConnection.PlayerInfo.Parse (player);
-							if (info.m_AllowDebugging) {
-								UnityPlayers[info.m_Guid] = info;
-								processes.Add (new ProcessInfo (info.m_Guid, info.m_Id));
-								++index;
+				if(Monitor.TryEnter (unityPlayerConnection)) {
```

Comment:
To avoid blocking the main thread.

https://github.com/Tak/MonoDevelop.Debugger.Soft.Unity/blob/master/UnitySoftDebuggerEngine.cs#L64